### ISSUE TITLE
About gpt_reward_test

### ISSUE THREAD

**Issue by ItGirls** [CarperAI/trlx#474]
### 🐛 Describe the bug

I find the result of the same input text under different batch_size will be different when run the gpt_reward_test with a reward model based on chatglm, of course the details of the implemention is different with your code for the padding side of chatglm is left.

**Comment by reciprocated** [comment]
Hi @ItGirls! Could you please give more context for this issue and also steps to reproduce it and a description of an expected behavior. Many thanks!

**Comment by ItGirls** [comment]
> I'm sorry,I find it's my mistake. But, I still want to mention that the code of reward model is not suit for LLM with left padding-side, for example the padding side of chat-glm is only allowed to be left.

**Comment by ItGirls** [comment]
in DataCollatorReward(74th line in train_reward_model_gptj.py), why 'batch["labels"] = torch.tensor([0] * len(data) + [1] * len(data))', is it wrong, cause chosen txt is before the rejected txt, so the right code may be ' batch["labels"] = torch.tensor([1] * len(data) + [0] * len(data)) '.

**Comment by maxreciprocate** [comment]
Sorry for the late reply. You're right, training code here assumes right-padding as well as in [autocrit](https://github.com/CarperAI/autocrit/blob/main/train_reward_model.py) right padding is assumed because of how `AutoModelForSequenceClassification` gathers logits. If chatglm still only permits left padding and if you need a reward model for it, we can work it out on the autocrit repository.
